### PR TITLE
Add documentation of Neato OAuth flow

### PR DESCRIPTION
**Pull request by Santobert**
## Proposed change

Add documentation of Neato OAuth2 config flow.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/44031
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/neato.markdown [hunk 1]
```diff
@@ -24,40 +25,38 @@ There is support for the following platform types within Home Assistant:
...
-Menu: *Configuration* -> *Integrations*
-
-Search for or select **Neato** from the list and configure the integration. You will need to enter your username and password and whether you are using a Neato or Vorwerk device.
-After that, all the entities will automatically show up in Home Assistant.
-
-## Setup the integration via `configuration.yaml`
+<div class='note'>
+You will have to enter an name, a description and your redirect URL.
+If homeassistant runs on <https://hass.example.com> your redirect URL would be <https://hass.example.com/auth/external/callback>.
+Please note that you must use a secure redirect URL (https://).
```
Comment (frenck) [inline source/_integrations/neato.markdown]: Having SSL is not a requirement of the OAuth2 specifications?

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/neato.markdown (hunk 1 above)
Comment (Santobert) [reply source/_integrations/neato.markdown]: Unfortunately neato requires that the redirect URL starts with https. I tried it and it doesn't work without ssl.

grafik

Comment (frenck) [reply source/_integrations/neato.markdown]: I hate that companies divert slightly and making their own protocols 😢

File: source/_integrations/neato.markdown [hunk 2]
```diff
@@ -24,40 +25,38 @@ There is support for the following platform types within Home Assistant:
 - **Switch** - allows you to enable or disable the schedule.
 - [**Vacuum**](#vacuum)
 
-To activate `neato` in your installation, you can set it up from the integration screen or add it to your `configuration.yaml` file.
+## Setup the integration
```
Comment (frenck) [inline source/_integrations/neato.markdown]:
```suggestion
## Configuration
```

File: source/_integrations/neato.markdown [hunk 3]
```diff
@@ -24,40 +25,38 @@ There is support for the following platform types within Home Assistant:
...
 
-## Setup the integration via the integrations screen
+1. Visit <https://developers.neatorobotics.com/applications> and create a new app.
```
Comment (frenck) [inline source/_integrations/neato.markdown]:
```suggestion
1. Visit [the Neato Developer Network](https://developers.neatorobotics.com/applications) and create a new app.
```

File: source/_integrations/neato.markdown [hunk 4]
```diff
@@ -24,40 +25,38 @@ There is support for the following platform types within Home Assistant:
...
-
-Search for or select **Neato** from the list and configure the integration. You will need to enter your username and password and whether you are using a Neato or Vorwerk device.
-After that, all the entities will automatically show up in Home Assistant.
-
-## Setup the integration via `configuration.yaml`
+<div class='note'>
+You will have to enter an name, a description and your redirect URL.
+If homeassistant runs on <https://hass.example.com> your redirect URL would be <https://hass.example.com/auth/external/callback>.
+Please note that you must use a secure redirect URL (https://).
+</div>
```
Comment (frenck) [inline source/_integrations/neato.markdown]:
```suggestion
<div class='note'>

You will have to enter a name, a description and your redirect URL.

If Home Assistant runs on `https://hass.example.com` your redirect URL would be `https://hass.example.com/auth/external/callback`.

Please note that you must use a secure redirect URL (`https://`).

</div>
```

**Review by frenck — CHANGES_REQUESTED** [review]

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Comment by Santobert** [comment]
Todo: mention check boxes

File: source/_integrations/neato.markdown [hunk 5]
```diff
@@ -24,40 +25,42 @@ There is support for the following platform types within Home Assistant:
 - **Switch** - allows you to enable or disable the schedule.
 - [**Vacuum**](#vacuum)
 
-To activate `neato` in your installation, you can set it up from the integration screen or add it to your `configuration.yaml` file.
+## Configuration
 
-## Setup the integration via the integrations screen
+1. Visit [the Neato Developer Network](https://developers.neatorobotics.com/applications) and create a new app.
```
Comment (dshokouhi) [inline source/_integrations/neato.markdown:30]: Lets make sure to mention the user needs to check all 3 scopes for the items below:

image

**Review by dshokouhi — CHANGES_REQUESTED** [review]
We should be a little more explicit here and also mention that they need provide the scopes for the application too.

File: source/_integrations/neato.markdown (hunk 5 above)
Comment (Santobert) [reply source/_integrations/neato.markdown:30]: Done. Thanks :)

**Review by dshokouhi — DISMISSED** [review]
Looks good to me :)

**Review by frenck — APPROVED** [review]
Thanks! 👍

**Pull request merged**